### ISSUE TITLE
Widget's "ViewWillAppear" never get called

### ISSUE THREAD

**Issue by zzz2016** [dji-sdk/Mobile-UXSDK-Beta-iOS#9]
`self.showDetailViewController(DefaultLayoutViewController(), sender: self)` ->  everything is ok.

`self.show(DefaultLayoutViewController(), sender: self)` -> widget's viewWillAppear will never be called.

**Comment by dji-lisafedane** [comment]
Thanks for reporting this - our engineers are looking into this now

**Comment by dji-lisafedane** [comment]
@zzz2016 - our engineers have some questions: 

1. What widget are you using?
2.  Are you trying to add the default view to your view controller? If so, you should use `install(in:)`

**Comment by zzz2016** [comment]
> @zzz2016 - our engineers have some questions:
> 
> 1. What widget are you using?
> 2. Are you trying to add the default view to your view controller? If so, you should use `install(in:)`

1. topBarWidget
2. I did it, but it doesn't work.

**Comment by zzz2016** [comment]
```
public func addRightWidgetArray(_ displayWidgets: [DUXBetaBaseWidget]) {
        self.suspendUpdates = true
        for aWidget in displayWidgets {
            // Build our tracking tuple for each widget so we can associate constraint.
            let isVisible = !(aWidget.view?.isHidden ?? true)   // The true means isHidden since there is no view
            let workTuple : LayoutInfoTuple = (aWidget, isVisible, nil, nil)
            allWidgetVisibilities[BarIndex.rightBarIndex.rawValue].append(workTuple)
            // Add an abserver to the hidden property so we can handle automatic adjustment when a widget disappears or reappears
            aWidget.view?.addObserver(self, forKeyPath: "hidden", options: .new, context: nil)

            addChild(aWidget)

            // And always have all widgets in the bar, even if invisible.
            self.rightBar.addArrangedSubview(aWidget.view)
            aWidget.view.layoutIfNeeded()
        }
        self.suspendUpdates = false
        self.updateUI()
} 
```
if i insert a line `addChild(aWidget)`， everything will be fine.

**Comment by dji-lisafedane** [comment]
@zzz2016 - thanks for your patience and reporting this to us.  we are looking into this now

**Comment by shane-dji** [comment]
I've been investigating this problem and am unable to reproduce it.
Steps I have tried in our sample app:
1. Removing the seque to show the DefaultLayoutViewController
2. Adding a IBAction to on the Default Layout button to perform:
   a) self.showDetailViewController(DefaultLayoutViewController(), sender: self) // This works as stated
   or
   b) self.show(DefaultLayoutViewController(), sender: self) // This causes the widget's viewWillAppear to be called .
3. I added a viewWillAppear to topBarWidget. It is called when self.show(DefaultLayoutViewController(), sender: self) 
4. I added a logging breakpoint to DUXBetaBatteryWidget which I then instantiated a new copy of and add to the top bar via  addLeftWidgetArray([DUXBetaBatteryWidget()])
5. The viewWillAppear for the battery widget is called successfully.

If you have any more information on which device/simulator this happens on and which version of iOS/Xcode, I would appreciate it.
